# Using this Package

## System Requirements
In general genome analysis carries relatively high demands for
compute, RAM, and disk I/O resources (but not for graphics resources, at
least not with the software used in this project). This project requires more than 128G of RAM for this purpose.

This project also makes use of various resource-intensive R packages.

Nonetheless, the necessary hardware is attainable in high-end
consumer-grade systems.

### Description of Development Systems
The following systems were used during development of this work:

1. System76 "Serval" mobile workstation
    * Intel i7-8700k CPU
    * 64GB RAM (DDR4-3000, non-ECC)
    * Samsung NVMe Pro SSD
    * Pop!_OS 17.10
2. PowerSpec G460 desktop computer
    * Intel i7-8700k CPU
    * 48GB RAM (DDR4-3200, non-ECC)
    * Intel M.2 SATA SSD
    * Windows 10 Pro ; Windows Subsystem for Linux, "WLinux" flavor

Note that WSL is still evolving and rough around the edges.  For example,
the "atom" editor has some bugs that prevent plug-in installation under WSL.

In Q4 2018, Apple announced a new Mac Mini lineup.  A fully upgraded Mini
seems to have specs comparable to the development systems and in theory
should be suitable, but no attempt has been made to evaluate this.  Note
that some adjustments to workflows may be necessary on MacOS.  For example,
Mac OS does not make the 'wget' utility available by default, nor do they
make it trivially easy to install (there are several options for
installing it, but they are not trivial in the sense that 'sudo apt-get install'
is trvial).

### Performance of Development Systems

System (1) is capable, starting from scratch, to download and compile
the STAR source code, download human genome data, align a reference
human genome, and translate 8 .fastq.gz files into .bam counterparts,
all in about 75 minutes.

System (2) is capable, starting from scratch, to 'make all' in about 4 hours.

The "Million of reads per hour" figures recorded in `*Log.final.out` files
generated by STAR, and summarized by R, indicate the following alignment
performance for the development systems:
```
          Min. 1st Qu.  Median    Mean 3rd Qu.    Max.
System 1 336.4   347.5   352.1   357.8   367.7   380.0
System 2 288.0   296.4   314.6   312.9   324.9   348.0
```

## DATASETS
Please download all datasets from the following weblinks and store them at
~/Download/
This project makes use of the following data files that must be fetched from remote sources:

### Xena DATA
https://pancanatlas.xenahubs.net/download/broad.mit.edu_PANCAN_Genome_Wide_SNP_6_whitelisted.gene.xena.gz
https://pancanatlas.xenahubs.net/download/EB++AdjustPANCAN_IlluminaHiSeq_RNASeqV2.geneExp.xena.gz
https://pancanatlas.xenahubs.net/download/TCGA_phenotype_denseDataOnlyDownload.tsv.gz
https://tcga.xenahubs.net/download/TCGA.PANCAN.sampleMap/Gistic2_CopyNumber_Gistic2_all_thresholded.by_genes.gz
https://toil.xenahubs.net/download/GTEX_phenotype.gz
https://toil.xenahubs.net/download/gtex_RSEM_Hugo_norm_count.gz
https://toil.xenahubs.net/download/TcgaTargetGTEX_phenotype.txt.gz
https://toil.xenahubs.net/download/TcgaTargetGtex_RSEM_Hugo_norm_count.gz
https://xenabrowser.net/datapages/?dataset=Survival_SupplementalTable_S1_20171025_xena_sp&host=https%3A%2F%2Fpancanatlas.xenahubs.net
### CPTAC DATA
#### CPTAC LUAD Proteomics
https://cptc-xfer.uis.georgetown.edu/publicData/Phase_III_Data/CPTAC_LUAD_S046/CPTAC_LUAD_Proteome_CDAP_Protein_Report.r1/CPTAC3_Lung_Adeno_Carcinoma_Proteome.tmt10.tsv
#### CPTAC LUAD Phosproteomics
https://cptc-xfer.uis.georgetown.edu/publicData/Phase_III_Data/CPTAC_LUAD_S046/CPTAC_LUAD_Phosphoproteome_CDAP_Protein_Report.r1/CPTAC3_Lung_Adeno_Carcinoma_Phosphoproteome.phosphosite.tmt10.tsv
#### CPTAC LUAD Sample Annotation
https://cptc-xfer.uis.georgetown.edu/publicData/Phase_III_Data/CPTAC_LUAD_S046/CPTAC_LUAD_metadata/S046_BI_CPTAC3_LUAD_Discovery_Cohort_Samples_r1_May2019.xlsx
#### CPTAC Clinical Data
https://cptc-xfer.uis.georgetown.edu/publicData/Phase_III_Data/CPTAC_LUAD_S046/CPTAC_LUAD_metadata/S046_BI_CPTAC3_LUAD_Discovery_Cohort_Clinical_Data_r1_May2019.xlsx


## LIBRARIES
Note that libraries are listed for convenience in "libraries.R".
The following command may be a useful way to install them all:
```
BiocManager::install(c("clusterProfiler",
                       "circlize",
                       "ComplexHeatmap",
                       "corrplot",
                       "data.table",
                       "dendextend",
                       "descr",
                       "dplyr",
                       "EnvStats",
                       "eulerr",
                       "factoextra",
                       "FactoMineR",
                       "forcats",
                       "forestmodel",
                       "Hmisc",
                       "ggfortify",
                       "ggplot2",
                       "ggpubr",
                       "ggsignif",
                       "ggthemes",
                       "glmnet",
                       "gplots",
                       "gridExtra",
                       "igraph",
                       "KEGG.db",
                       "lemon",
                       "limma",
                       "missMDA",
                       "org.Hs.eg.db",
                       "pheatmap",
                       "plotmo",
                       "RColorBrewer",
                       "ReactomePA",
                       "readr",
                       "readxl",
                       "reshape2",
                       "rgl",
                       "scales",
                       "survival",
                       "survivalAnalysis",
                       "survMisc",
                       "survminer",
                       "tidyverse",
                       "vcd",
                       "vip"
), update = TRUE, ask = FALSE)
```

## OUTPUT
The script will save all the generated plots into the following output folder.
Please generate the output directory.
```
mkdir -p ~/Documents/EIF_output/{CNV,Expression,PCA,KM,Cox,Heatmap,CPTAC}
```

## Run the script.
1. Install R (3.6) and RStudio at the recommended Systems.
2. Download all datasets from the listed web links above and store them under folder ~/Download/
3. Install all the R packages listed above.
4. Create an output folder ~/Documents/EIF_output/{CNV,Expression,PCA,KM,Cox,Heatmap,CPTAC}
5. Open the EIFanalysis.R file, and run it with the source button.
6. The EIFanalysis.R will produce all the figures from the EIF manuscript.
